flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -243,24 +248,33 @@ protected NMClientAsync createAndStartNodeManagerClient(YarnConfiguration yarnCo
 
 		// domain objects for test purposes
 		final ResourceProfile resourceProfile1 = ResourceProfile.UNKNOWN;
+		final WorkerResourceSpec workerResourceSpec;
+
+		final Resource containerResource;
 
 		public String taskHost = "host1";
 
 		final TestingYarnNMClientAsync testingYarnNMClientAsync;
 
 		final TestingYarnAMRMClientAsync testingYarnAMRMClientAsync;
 
+		int containerIdx = 0;
+
 		/**
 		 * Create mock RM dependencies.
 		 */
 		Context() throws Exception {
-			this(flinkConfig);
+			this(flinkConfig, null);
 		}
 
-		Context(Configuration configuration) throws  Exception {
-			final SlotManager slotManager = SlotManagerBuilder.newBuilder()
-				.setDefaultWorkerResourceSpec(YarnWorkerResourceSpecFactory.INSTANCE.createDefaultWorkerResourceSpec(configuration))
-				.build();
+		Context(Configuration configuration, @Nullable SlotManager slotManager) throws  Exception {
+
+			workerResourceSpec = YarnWorkerResourceSpecFactory.INSTANCE.createDefaultWorkerResourceSpec(configuration);
+			if (slotManager == null) {
+				slotManager = SlotManagerBuilder.newBuilder()
+					.setDefaultWorkerResourceSpec(workerResourceSpec)
+					.build();
+			}
```

Comment:
It requires `workerResourceSpec` for creating the default `slotManager`.
I was trying to avoid creating `workerResourceSpec` twice, in both constructors.

Reply:
I think it would be fine to add a `SlotManager.getDefaultWorkerResourceSpec()`.